File: src/main/java/com/qml/usecases/Ex01.java
package com.qml.usecases;

import static org.superbiz.tf.QMLContext.createContext;
import static org.superbiz.tf.QMLContext.value;

public class Ex01 {
    // x = tf.Variable(3, name="x")
    //y = tf.Variable(4, name="y")
    //f = x*x*y + y + 2
    //
    //with tf.Session() as sess:
    //    # x.initializer.run()
    //    # y.initializer.run()
    //    init = tf.global_variables_initializer()
    //    init.run()
    //    result = f.eval()
    //    print(result)

//    public static void main(String[] args) {
//        try (QMLContext tf = createContext("TensorFlow")) {
//            TF<Variable> x = tf.variable(value(3), named("x"));
//            TF<Variable> y = tf.variable(value(4), named("y"));
//            TF<BasicOperation.Add> add = x.add(y);
//
//            tf.run(tf.globalVariablesInitializer());
//            Float result = tf.fetch(add);
//        }
//
//    }
}

File: src/main/java/com/qml/usecases/ex01.py
import tensorflow as tf

# x = tf.constant(3.1, name="x")
# #y = tf.Variable(4, name="y")
# f = tf.square(x)

#one = tf.constant(1.0, shape=[1], name='one')
x = tf.constant([1.8, 2.2], dtype=tf.float32)
f = tf.negative(x)

with tf.Session() as sess:
    init = tf.global_variables_initializer()
    init.run()
    result = f.eval()
    print(result)

    graph_def = tf.get_default_graph().as_graph_def()
    tf.train.write_graph(graph_def, '../../../../../../src/main/resources/com/qml/usecases', 'ex01.pb', False)

    #tf.train.write_graph(graph_def, '/tmp', 'ex01.pb', False)
